bugsnag-plugin-react-native/src/main/java/com/bugsnag/android/BugsnagReactNativePlugin.kt

```diff
@@ -52,13 +52,25 @@ class BugsnagReactNativePlugin : Plugin {
         notifier.dependencies.add(Notifier()) // depend on bugsnag-android
     }
 
+    private fun ignoreJavaScriptExceptions() {
+        this.client.addOnError(OnErrorCallback { event ->
+            event.errors[0].errorClass != "com.facebook.react.common.JavascriptException"
+        })
+    }
+
     override fun unload() {}
 
     @Suppress("unused")
     fun configure(env: Map<String, Any?>?): Map<String, Any?> {
         requireNotNull(env)
         updateNotifierInfo(env)
 
+        // JS exceptions are caught by the JS layer and passed to the dispatch method
+        // in this plugin. When a JS exception crashes the app, we get a duplicate
+        // native exception for the same error so we ignore those once the JS layer
+        // has started.
+        ignoreJavaScriptExceptions()
```

Comment:
I wanted to ensure it didn't take effect until the JS layer had come into effect. Say there is a JS crash before the JS part of the notifier is configured, putting it in `load` makes the chance of reporting that effectively zero.

Would you be happy for me to adjust this so that it checks a stateful flag before calling `ignoreJavaScriptExceptions()`?

Reply:
Ah, good point. A stateful flag seems the way forward in that case.